nss-day-cohort-60/national-parks-api
# Md natural attractions

Description:
1. Added new sql helper function called get_all_by_param to handle get all by one query parameter into sql_helper.py
2. Implemented  get_all_by_param function into get_wildlife_by_park_id and get_natural_attractions_by_park_id
3. Added create, update, and delete functions into wildlife_views.py and natural_attractions_views.py
4. Updated method_mapper in method_mapper.py with all functions in wildlife_views.py and natural_attractions_views.py
5. Added to def retrieve in repository.py to check id the request was of the type resource?parameter=value and if it is, to use the appropriate function from method_mapper

Steps to test:
1. Fetch the md-natural-attractions branch and switch to it 
2. Start/restart debugger/server
3. Open Postman and use to make the following requests:
- Send a GET request for all existing wildlife by park_id (example: http://localhost:8088/wildlife?park_id=1)
Verify that 200 status code and that all wildlife from that particular park are displayed with the park name
- Repeat process for get_natural_attractions_by_park_id
4. Use Postman to send a POST, PUT and DELETE request for wildlife and natural_attractions
Examples:
a. Send a DELETE request for an existing wildlife (e.g. http://localhost:8088/wildlife/3)
- Verify that 204 status code is received
- Request all wildlife and verify it was deleted
-Repeat this process for natural_attractions
b.POST a new wildlife type to http://localhost:8088/wildlife
{  
"name": "test me" ,
"information": "lorem ipsum"
"wildlife_group_id": 3
"image":"https://res.cloudinary.com/dcuwovsbv/image/upload/v1674244557/Wildlife/BOREAL-TOAD-ANAXYRUS-BOREAS-2018-339_wvrokb.webp"
}  

Verify that you get a single dictionary back.  
GET all wildlife and make sure the new type has been added
-Repeat this process for natural_attractions

c.Use PUT to make a change to an existing wildlife type by specifying the id  (example: http://localhost:8088/wildlife/2)
{  
"name": "test me" ,
"information": "NEW CHANGE"
"wildlife_group_id": 3
"image":"https://res.cloudinary.com/dcuwovsbv/image/upload/v1674244557/Wildlife/BOREAL-TOAD-ANAXYRUS-BOREAS-2018-339_wvrokb.webp"
}  

Verify that you get a single dictionary back.  
GET all wildlife and make sure the change has been made
-Repeat this process for natural_attractions
